src/main/java/thaumic/tinkerer/common/item/ItemCleansingTalisman.java

```diff
@@ -60,6 +59,11 @@ public class ItemCleansingTalisman extends ItemBase implements IBauble {
 
     private IIcon enabledIcon;
 
+    private static final int EFFECT_BURNING = -1;
+    // 0 = no effect, otherwise either EFFECT_BURNING or the ID of the effect removed.
+    private int lastEffectRemoved = 0;
+    private final Collection<Integer> permanentEffects = new ArrayList<>();
```

Comment:
The issue with current implementation is that while it perfectly runs in singleplayer, one will break in multiplayer. Because there is *only one instance* of `Item` - there's *only one instance* of this collection. This means that *all* players will use the same collection - this list will store potions of *all* players at the same time.
Consider to changing collection type to `HashMap<UUID, Integer>` to store data per-player.

Reply:
Oops! This I did not know, thanks for catching it. Should be fixed now.